### ISSUE TITLE
Serialization issue with scalapb.TypeMapper

### ISSUE THREAD

**Issue by joagain** [scalapb/ScalaPB#1080]
Hi,

My issue is similar to https://github.com/scalapb/ScalaPB/issues/841 and https://github.com/scalapb/ScalaPB/issues/884
I understand it should be fixed in 0.9.3, but I am not sure why I still encounter the error.
My protobuf object contains a map.

```
	- object not serializable (class: scalapb.TypeMapper$$anon$1, value: scalapb.TypeMapper$$anon$1@4b96b0bd)
```

Here is the code:
```
    import scalapb.spark.ProtoSQL
    import spark.implicits.StringToColumn
    val parseData = ProtoSQL.udf { bytes: Array[Byte] => MyProtobufObject.parseFrom(bytes) }
    df.withColumn("parsed_value", parseData($"value"))
```

My dependencies:
```
compile group: 'com.thesamet.scalapb', name: 'sparksql-scalapb_2.11', version: '0.9.3'
compile group: 'com.thesamet.scalapb', name: 'scalapb-runtime_2.11', version: '0.9.8' // I also tried with 0.9.3
...
compileOnly group: 'org.apache.spark', name: 'spark-sql_2.11', version: '2.4.7'
```

How I generate the Proto classes:
```
protobuf {
    protoc {
        artifact = "com.google.protobuf:protoc:3.12.2"
    }
    plugins {
        scalapb {
            artifact = "com.thesamet.scalapb:protoc-gen-scala:0.9.3:unix@sh"
        }
    }
    ...
}
```

@thesamet Would you have any guidance? Thank you very much!

**Comment by thesamet** [comment]
Have you tried regenerating the code with protoc-gen-scala 0.9.8? Only changing the version of runtime wouldn't be sufficient. The relevant fix was committed for 0.9.6 (see https://github.com/scalapb/ScalaPB/commit/5a8925b3d7b820ffb533909f86c7134af3e56a2a) and it makes sure the `@transient` annotation is written before typemappers so this error is not triggered.

**Comment by joagain** [comment]
@thesamet Thank you so much, I generated with `0.9.8` and overcame this error. Again, thank you. I am closing this ssue.
